core/modules/widgets/transclude.js

```diff
@@ -60,8 +61,10 @@ TranscludeWidget.prototype.execute = function() {
 						}),
 		parseTreeNodes = parser ? parser.tree : this.parseTreeNode.children;
 	// Set context variables for recursion detection
-	var recursionMarker = this.makeRecursionMarker();
-	this.setVariable("transclusion",recursionMarker);
+	if(this.recursionMarker === "yes") {
+		var recursionMarker = this.makeRecursionMarker();
+		this.setVariable("transclusion",recursionMarker);
+	}
 	// Check for recursion
 	if(parser) {
 		if(this.parentWidget && this.parentWidget.hasVariable("transclusion",recursionMarker)) {
```

Comment:
Note that here we use the recursionMarker variable again, so the conditional above should only be around the setVariable, and not the var.

Reply:
@Jermolene good catch, thank you.